channels/drive/client/drive_main.c

```diff
@@ -867,6 +868,7 @@ UINT drive_register_drive_path(PDEVICE_SERVICE_ENTRY_POINTS pEntryPoints,
 out_error:
 	MessageQueue_Free(drive->IrpQueue);
 	ListDictionary_Free(drive->files);
+	free(drive->path);
```

Comment:
This looks broken in context. Should use ```drive_free``` to remove duplication (and possible missing cleanup)